base/applications/mspaint/miniature.cpp

```diff
@@ -32,23 +33,72 @@ HWND CMiniatureWindow::DoCreate(HWND hwndParent)
     return Create(hwndParent, rc, strTitle, style, WS_EX_PALETTEWINDOW);
 }
 
+LRESULT CMiniatureWindow::OnMove(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (IsWindowVisible() && !IsIconic() && !IsZoomed())
+    {
+        registrySettings.ThumbXPos = GET_X_LPARAM(lParam);
+        registrySettings.ThumbYPos = GET_Y_LPARAM(lParam);
+    }
+    return 0;
+}
+
+LRESULT CMiniatureWindow::OnSize(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (IsWindowVisible() && !IsIconic() && !IsZoomed())
+    {
+        registrySettings.ThumbWidth = GET_X_LPARAM(lParam);
+        registrySettings.ThumbHeight = GET_Y_LPARAM(lParam);
+    }
+    return 0;
+}
+
 LRESULT CMiniatureWindow::OnClose(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
 {
     ShowWindow(SW_HIDE);
     registrySettings.ShowThumbnail = FALSE;
     return 0;
 }
 
+LRESULT CMiniatureWindow::OnEraseBkgnd(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    return TRUE; /* Avoid flickering */
+}
+
 LRESULT CMiniatureWindow::OnPaint(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
 {
     RECT rc;
     GetClientRect(&rc);
 
+    // Start painting
     PAINTSTRUCT ps;
     HDC hDC = BeginPaint(&ps);
-    StretchBlt(hDC, 0, 0, rc.right, rc.bottom,
-               imageModel.GetDC(), 0, 0, imageModel.GetWidth(), imageModel.GetHeight(),
-               SRCCOPY);
+    if (!hDC)
+        return 0;
+
+    // Use a memory bitmap to reduce flickering
```

Comment:
Does this indeed give a visual difference when run on Windows?

Reply:
Yep. The miniature window does flickering.